web/src/main/java/org/springframework/security/web/csrf/DefaultCsrfToken.java

```diff
@@ -72,6 +81,73 @@ public String getParameterName() {
 	 * @see org.springframework.security.web.csrf.CsrfToken#getToken()
 	 */
 	public String getToken() {
-		return this.token;
+		Random randomSize = new Random();
```

Comment:
I added random size to add more randomness to the token. But I think will have issue with `matches` method logic since like you mentioned required to use constant time comparison.

Will remove this and fix it to 5 instead.

Reply:
I'm thinking a larger value is better to reduce the likelihood that we get duplicate values. I'd suggest having the value be the same size as the actual token.